### PR TITLE
Misc. no-op refactoring to prep for meter param override invalidation

### PR DESCRIPTION
**Pull request by pattyshack**
1. move ModifiedSetsInvalidator from programs/ to environment/ (+ some renaming)
2. replace ModifiedSetsInvalidator with TransactionInvalidator interface where possible.
3. add nil check to AddInvalidator (this was a bug)
4. made a bunch of fields/structs public to facilitate testing
5. use RegisterID instead of string as test ledger's keys
6. move environment tests that uses mocks to environment_test package to avoid circular dependency

gh: #3102

### CODE REVIEW

File: fvm/utils/test.go [hunk 1]
```diff
@@ -105,51 +100,43 @@ func (v *SimpleView) Delete(owner, key string) error {
...
 }
 
 // NewMapLedger returns an instance of map ledger (should only be used for testing)
 func NewMapLedger() *MapLedger {
 	return &MapLedger{
-		Registers:       make(map[string]flow.RegisterEntry),
-		RegisterTouches: make(map[string]bool),
-		RegisterUpdated: make(map[string]bool),
+		Registers:       make(map[flow.RegisterID]flow.RegisterEntry),
+		RegisterTouches: make(map[flow.RegisterID]bool),
```
Comment (SaveTheRbtz) [inline fvm/utils/test.go:112]: nit: map[flow.RegisterID]struct{}

File: fvm/utils/test.go [hunk 2]
```diff
@@ -105,51 +100,43 @@ func (v *SimpleView) Delete(owner, key string) error {
...
+	RegisterUpdated map[flow.RegisterID]bool
 }
 
 // NewMapLedger returns an instance of map ledger (should only be used for testing)
 func NewMapLedger() *MapLedger {
 	return &MapLedger{
-		Registers:       make(map[string]flow.RegisterEntry),
-		RegisterTouches: make(map[string]bool),
-		RegisterUpdated: make(map[string]bool),
+		Registers:       make(map[flow.RegisterID]flow.RegisterEntry),
```
Comment (SaveTheRbtz) [inline fvm/utils/test.go:111]: nit: map[flow.RegisterID]flow.RegisterValue

**Review by SaveTheRbtz — APPROVED** [review]
Thanks for cleaning up `map[string]` stuff in test!!

**Review by janezpodhostnik — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**